server/src/Main.hs

```diff
@@ -68,28 +67,60 @@ setAuthenticatedAs db userHandle = do
     "Set-Cookie"
     (LText.decodeUtf8 (Builder.toLazyByteString (Cookie.renderSetCookie setCookie)))
 
-getAuthenticatedUser :: Database.Connection -> Scotty.ActionM (Maybe M.UserHandle)
-getAuthenticatedUser db = runMaybeT $ do
+getAuthToken :: MaybeT Scotty.ActionM Database.AuthToken
+getAuthToken = do
   cookieHeader <- MaybeT $ Scotty.header "cookie"
   let cookies = Cookie.parseCookies $ LBS.toStrict $ LText.encodeUtf8 cookieHeader
   sessionCookie <- MaybeT . pure $ lookup "auth-token" cookies
-  token <- MaybeT . pure $ either (const Nothing) (Just . Database.AuthToken) $ Base64.decodeUnpadded sessionCookie
+  MaybeT . pure $ either (const Nothing) (Just . Database.AuthToken) $ Base64.decodeUnpadded sessionCookie
+
+getAuthenticatedUser :: Database.Connection -> Scotty.ActionM (Maybe M.UserAccountName)
+getAuthenticatedUser db = runMaybeT $ do
+  token <- getAuthToken
   MaybeT $
-    liftIO $
+    Scotty.liftAndCatchIO $
       Database.withTransaction db $ \tx ->
         Database.getAuthTokenUser tx token
 
+logout :: Database.Connection -> Scotty.ActionM ()
+logout db = do
+  runMaybeT getAuthToken >>= \case
+    Nothing -> pure ()
+    Just token -> do
+      Scotty.liftAndCatchIO $
+        Database.withTransaction db $ \tx ->
+          Database.removeAuthToken tx token
+
+  let setCookie =
+        Cookie.defaultSetCookie
+          { Cookie.setCookieName = "auth-token",
+            Cookie.setCookieValue = "",
+            Cookie.setCookieSameSite = Just Cookie.sameSiteStrict,
+            Cookie.setCookieSecure = True,
+            Cookie.setCookieHttpOnly = True,
+            Cookie.setCookiePath = Just "/",
+            Cookie.setCookieMaxAge = Just 0
+          }
+  Scotty.setHeader
+    "Set-Cookie"
+    (LText.decodeUtf8 (Builder.toLazyByteString (Cookie.renderSetCookie setCookie)))
+
 app :: M.Origin -> M.RpIdHash -> Database.Connection -> PendingOps -> ScottyM ()
 app origin rpIdHash db pending = do
   Scotty.middleware (staticPolicy (addBase "dist"))
+  Scotty.get "/" $ do
+    getAuthenticatedUser db >>= \case
+      Nothing -> Scotty.redirect "unauthenticated.html"
+      Just _ -> Scotty.redirect "authenticated.html"
```

Comment:
Non authorized users should also be redirected to `unauthenticated.html`, when they visit `authenticated.html`.

Reply:
That is a bit of a pain to set up, because `[un]authenticated.html` is handled by the above static redirects. I'd rather keep it like this for simplicity

Reply:
Is it instead possible to have an error message show when you visit `authenticated.html` without being authenticated?

Reply:
I've now made it so that the javascript redirects you to unauthenticated.html when you're not authenticated

Reply:
That was fairly easy to add

Reply:
The problem I have is that it feels we are leaking information by allowing unauthenticated users to visit that page. Surely, we don't, it's an example and everything, but it does feel like it. Using the JS for that doesn't solve this issue of course.

If you feel it's a non-issue, feel free resolve this conversation.